docker/docker_APP/Dockerfile

```diff
@@ -0,0 +1,41 @@
+#---------------------------------------------------------------------------
+# Dockefile to build Docker Image of application container for spring_APP running on Ubuntu
+#---------------------------------------------------------------------------
+
+# ======== CLONE SRC ==========
+FROM alpine/git:v2.26.2 as git
+
+# Setting vars
+# Define URL via --build-arg
+ENV PROJECT_DIR /springapp/spring-petclinic
+ARG URL 
+
+# Clone repo
+RUN mkdir /springapp && \
+cd /springapp && \
+git clone $URL
+
+# ========= BUILDER ========
+
+FROM openjdk:8-jdk-alpine as builder
+# COPY src and .m2
+ENV PROJECT_DIR /springapp/spring-petclinic
+RUN mkdir ~/.m2 
```

Comment:
The more RUNs you have the more layers it will produce in image. Try to keep as less layers as possible, but keep Dockerfile readable.